src/hooks/useWallpaper.jsx

```diff
@@ -18,46 +23,79 @@ const useWallpaper = () => {
       }
       try {
         setFetchStatus('loading')
-        const binary = await localForage.getItem('customWallpaper')
-        if (binary) {
-          setBinaryCustomWallpaper(binary)
-        }
-        const response = await client
-          .collection('io.cozy.files')
-          .getDownloadLinkByPath(homeConfig.customWallpaperPath)
-        setWallpaperLink(response)
-        setFetchStatus('loaded')
-        const fetchBinary = await fetch(response)
-        const blob = await fetchBinary.blob()
-        const reader = new FileReader()
-        reader.readAsDataURL(blob)
-        reader.onloadend = async () => {
-          const base64data = reader.result
-          setBinaryCustomWallpaper(base64data)
-          await localForage.setItem('customWallpaper', base64data)
-          localStorage.setItem('hasCustomWallpaper', true)
+        const link = await localForage.getItem('customWallpaper')
+        if (link) {
+          setWallpaperLink(link)
+          setFetchStatus('loaded')
+          return
         }
       } catch (error) {
         await localForage.removeItem('customWallpaper')
         localStorage.setItem('hasCustomWallpaper', false)
         setWallpaperLink(cozyDefaultWallpaper)
-        setBinaryCustomWallpaper(null)
         setFetchStatus('failed')
       }
     }
     fetchData()
   }, [client, cozyDefaultWallpaper])
 
+  const setWallpaperLinkAndStore = async link => {
+    setWallpaperLink(link)
+    if (link === cozyDefaultWallpaper) {
+      await localForage.removeItem('customWallpaper')
+      localStorage.setItem('hasCustomWallpaper', false)
+    }
+    localStorage.setItem('hasCustomWallpaper', true)
+    await localForage.setItem('customWallpaper', link)
+  }
+
+  /**
+   * Resets the wallpaper to the Cozy default wallpaper.
+   *
+   * This function sets the wallpaper link back to the default wallpaper,
+   * removes any custom wallpaper reference from localForage storage,
+   * and updates the localStorage flag indicating that there is no custom wallpaper.
+   *
+   * @async
+   * @returns {Promise<void>}
+   */
+  const returnToDefaultWallpaper = async () => {
+    setWallpaperLink(cozyDefaultWallpaper)
+    await localForage.removeItem('customWallpaper')
+    localStorage.setItem('hasCustomWallpaper', false)
+  }
+
+  /**
+   * Clears the custom wallpaper.
+   *
+   * This function deletes the binary wallpaper file, clears all related caches,
+   * and resets the wallpaper to the Cozy default wallpaper. It also updates
+   * localForage and localStorage to reflect that no custom wallpaper is currently set.
+   *
+   * @async
+   * @returns {Promise<void>}
+   */
+  const clearCustomWallpaper = async () => {
+    // Delete the binary file, clear caches and return to default wallpaper
+    await clearBinary()
+    setWallpaperLink(cozyDefaultWallpaper)
+    await localForage.removeItem('customWallpaper')
+    localStorage.setItem('hasCustomWallpaper', false)
```

Comment:
```suggestion
returnToDefaultWallpaper()
```

this code is the same as the `returnToDefaultWallpaper` so why not use it